messaging/filter/service.go

```diff
@@ -48,23 +49,33 @@ type Chat struct {
 	Negotiated bool `json:"negotiated"`
 }
 
+type Messages struct {
+	Chat     *Chat              `json:"chat"`
+	Messages []*whisper.Message `json:"messages"`
+	Error    error              `json:"error"`
+}
+
 type Service struct {
-	whisper     *whisper.Whisper
-	secret      *sharedsecret.Service
-	chats       map[string]*Chat
-	persistence Persistence
-	mutex       sync.Mutex
-	keys        map[string][]byte
+	whisper       *whisper.Whisper
+	secret        *sharedsecret.Service
+	chats         map[string]*Chat
+	persistence   Persistence
+	mutex         sync.Mutex
+	keys          map[string][]byte
+	quit          chan struct{}
+	onNewMessages func([]*Messages)
```

Comment:
This worries me a bit that we do any message retrieval/processing in `filter`. Why do we need this?

Reply:
In status-react we poll through web3 for each filter, this changed the behavior so that the polling is done at status-go level, and there's no interaction with web3 (or whisper) anymore in status-react. Instead a signal is sent with multiple messages from different filters, for performance reasons (roughly 50% performance improvement).
We can change the interface that I don't mind, as long as it's compatible with status-react.